eventnotifier/http/http.go

```diff
@@ -266,3 +211,129 @@ func (n *Notifier) marshalUserGroups(groups []string) (string, error) {
 
 	return string(groupsBytes), nil
 }
+
+func (n *Notifier) notifyDisruption(dis v1beta1.Disruption, event corev1.Event, notifType types.NotificationType) error {
+	logger := n.logger
+
+	logger.With("disruptionName", dis.Name, "disruptionNamespace", dis.Namespace)
+
+	userInfo, err := dis.UserInfo()
+	if err != nil {
+		n.logger.Warnw("http notifier: no user info in disruption", "error", err)
+	}
+
+	notifierEvent := n.buildEvent(&dis, userInfo, event, notifType, logger)
+
+	disruptionStr, err := json.Marshal(dis)
+	if err != nil {
+		return fmt.Errorf("http notifier: couldn't marshal disruption: %w", err)
+	}
+
+	notifierEvent.DisruptionEvent = &DisruptionEvent{
+		TargetsCount:   len(dis.Status.TargetInjections),
+		DisruptionName: dis.Name,
+		Disruption:     string(disruptionStr),
+	}
+
+	jsonNotif, err := jsonapi.Marshal(&notifierEvent)
+	if err != nil {
+		return fmt.Errorf("http notifier: couldn't marshal notification: %w", err)
+	}
+
+	logger.With("eventType", event.Type, "message", notifierEvent.EventMessage)
+
+	return n.emitEvent(jsonNotif, logger)
+}
+
+func (n *Notifier) notifyDisruptionCron(disruptionCron v1beta1.DisruptionCron, event corev1.Event, notifType types.NotificationType) error {
```

Comment:
Why did we choose to implement disruption cron events in http notifier but not the datadog notifier? I don't think the api we have listening for disruptions is yet prepared for disruption crons

Reply:
In order to reduce the size of the PR I just add the support of `DisruptionCron` for the **http notifier**. Indeed, you make a good point. I should ensure the old fields are still there to not break the existing disruption registering, but the goal of my work is to send `DisruptionCron` to our API.

Reply:
we should create a jira ticket to support all types of notifiers

Reply:
I think we're only going to get errors from any crons we send to the api, as it isn't prepared to listen for those yet. Could we change the structure of our config to enable/disable notifications for each notifier type for each resource type separately?

i.e., instead of
```
http:
  enabled: true
```
could it look more like
```
http:
  enabled:
    disruptions: true
    disruptioncrons: false
```
? wdyt @diyarab @clairecng @aymericDD

Reply:
I have already tested functionally the new payload of the callback and it works with the chaos-api so no need to add an extra configuration 👍

Reply:
It feels uncomfortable to merge code to main, ungated by a config value, that we expect not to work? It's not always true (we ship bugs, ofc), but ideally any commit on main should be usable for a release